rose/server/score.py

```diff
@@ -19,62 +17,72 @@ def process(players, track):
     # First handle right and left actions, since they may change in_lane
     # status, used for resolving collisions.
 
-    for player in six.itervalues(players):
+    for player in players.values():
         if player.action == actions.LEFT:
             if player.x > 0:
                 player.x -= 1
         elif player.action == actions.RIGHT:
             if player.x < config.matrix_width - 1:
                 player.x += 1
 
-    # Now handle obstacles, preferring players in their own lane.
+        obstacle = track.get(player.x, player.y)
 
-    sorted_players = sorted(six.itervalues(players),
-                            key=lambda p: 0 if p.in_lane() else 1)
-    positions = set()
+        if obstacle == obstacles.NONE:
+            # Move forlward
+            player.score += config.score_move_forward
 
-    for player in sorted_players:
```

Comment:
I don't see how the change is correct when you don't process players in order.

The purpose of processing in order is to give priority to some players, so they
can acquire the next position before the other players. In the past we use the 
player response time to give priority to faster players but this was removed since
players do not use the same machines on the same network.

Reply:
> The purpose of processing in order is to give priority to some players

this is not documented in https://github.com/RedHat-Israel/ROSE/tree/master/examples

the documented rule is:

> NOTE : If you are driving out of your lane and have a collision with the lane owner, you will punished by -10 points and be moved to other free place.

The purpose of this PR is to align the code with the documentation,
So the change is correct to follow the documentation, it removes the undocumented logic:

change from checking the priority by sorting using undocumented criteria, 
to the explicitly documented rule of giving priority to the car that is in it's lane:
"If you are driving out of your lane"

When we update the documentation and change the collision rules (i don't think we should) we can update the code accordingly